shared/actions/chat2/index.js

```diff
@@ -1711,7 +1710,34 @@ function* attachmentDownload(action: Chat2Gen.AttachmentDownloadPayload) {
 
   // Download it
   const destPath = yield Saga.call(downloadFilePath, message.fileName)
-  yield Saga.call(downloadAttachment, destPath, conversationIDKey, message, ordinal)
+  yield Saga.call(downloadAttachment, destPath, message)
+}
+
+function* attachmentFullscreenNext(action: Chat2Gen.AttachmentFullscreenNextPayload) {
```

Comment:
if this is actionToAction you get state passed in and you don't need to call select

Reply:
you can still return a full blown generator in a call
```
const attachmentFullscreenNext = (state: TypedState, action: Chat2Gen.AttachmentFullscreenNextPayload) => Saga.call(function*() {
...
})
```

Reply:
If I do that then what do I gain over what is there now? Just having the state parameter?

Reply:
eventually we'll deprecate the safeTake*. this newer api helps enforce better practices but still gives you an escape hatch if you need to do more complicated async ops